common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -1417,6 +1417,13 @@ def save_problem(self, data):
             }
 
         self.lcp.student_answers = answers
+        answer_keys = [key for key in answers]
+        for answer_key in answer_keys:
+            # Reset the correctness property to default(None) state when attempted wrong
+            if self.lcp.correct_map.get_property(answer_key, 'correctness', None) == 'incorrect':
```

Comment:
This does not affect the student grading, you can check [here](http://problemsave.sandbox.edx.org/courses/course-v1:mushtaq+101+2015/progress/4/). You can try attempting wrong, and the save another choice and see the progress page shows the expected grades. You can then attempt a correct choice and then see progress page showing correct grades.

Also, I addressed your comments above, and now bug you mentioned in [above](#discussion_r38965811) comment is addressed as well; see screenshot below. Previously, I thought, there may be no need to save a choice, if user had submitted a correct answer.




<img width="850" alt="screen shot 2015-09-09 at 5 25 36 pm" src="https://cloud.githubusercontent.com/assets/6991154/9761713/d5cc6908-5717-11e5-8473-b3be5f170685.png">